codalab/worker_manager/kubernetes_worker_manager.py

```diff
@@ -131,27 +123,29 @@ def start_worker_job(self) -> None:
                         },
                         'volumeMounts': [
                             {'name': 'dockersock', 'mountPath': '/var/run/docker.sock'},
-                            {'name': 'workdir', 'mountPath': work_dir},
                         ],
                     }
                 ],
-                'volumes': [
-                    {'name': 'dockersock', 'hostPath': {'path': '/var/run/docker.sock'}},
-                    {'name': 'workdir', 'hostPath': {'path': work_dir}},
-                ],
+                'volumes': [{'name': 'dockersock', 'hostPath': {'path': '/var/run/docker.sock'}},],
```

Comment:
@teetone Can you explain how it was working earlier? I actually depend on this functionality (mounting `workdir`) for https://github.com/codalab/codalab-worksheets/pull/4090

Reply:
We're still mounting `workdir`. See the code below.

Reply:
Ok. Can we directly put the logic in this part of the code for clarity (rather than calling `.append()` later)?

Reply:
Are you suggesting using a couple of in-line if expressions? A separate if-else block seems cleaner, but I can make that change if you think it's clearer.

Reply:
Yeah I think in-line expressions are clearer, you can concatenate arrays. Here's how I did it in #4090 -- https://github.com/codalab/codalab-worksheets/blob/f35fa7519d11752d57df5216bdac8369447115d9/codalab/worker/runtime/kubernetes_runtime.py#L116-L133.

What do you think? Does this look cleaner?